modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/WeightedSampling.scala

```diff
@@ -0,0 +1,44 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.MonadThrow
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+
+object WeightedSampling {
+
+  type ProbabilityRefinement = Interval.ClosedOpen[0.0, 1.0]
+  type ProbabilityRefined = Double Refined ProbabilityRefinement
```

Comment:
`type Probability`

Reply:
Changed `ProbabilityRefined` to `Probability`.